mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSLanguageRepository.kt

```diff
@@ -40,13 +54,20 @@ class MPSLanguageRepository(private val repository: SRepository) : ILanguageRepo
         return MPSConcept(MetaAdapterFactory.getAbstractConcept(conceptDescriptor))
     }
 
-    override fun getAllConcepts(): List<IConcept> {
-        val result = mutableListOf<IConcept>()
-        LanguageRegistry.getInstance(repository).withAvailableLanguages { language ->
-            result.addAll(language.identity.concepts.map { MPSConcept(it) }.toList())
-        }
-        return result
-    }
+    fun getConcept(uid: String): SAbstractConcept? = concepts[uid]?.concept
+
+    override fun getAllConcepts(): List<IConcept> = concepts.values.toMutableList()
 
     override fun getPriority(): Int = 1000
+
+    override fun equals(other: Any?) =
+        if (this === other) {
+            true
+        } else if (javaClass != other?.javaClass) {
+            false
+        } else {
+            repository == (other as MPSLanguageRepository).repository
+        }
+
+    override fun hashCode() = repository.hashCode()
```

Comment:
In case the concept cache is dropped, we could just make this a `data class` to let kotlin handle these methods.

Reply:
fixed in[976691a](https://github.com/modelix/modelix.core/pull/237/commits/976691a5cbd5356adf6fe455c15ca77b4103575f)